src/main/java/com/mongodb/spark/sql/connector/schema/RowToBsonDocumentConverter.java

```diff
@@ -226,4 +241,17 @@ private BsonValue processString(final String data) {
     }
     return new BsonString(data);
   }
+
+  private boolean parseJsonData(final String data) {
+    switch (convertJson) {
+      case FALSE:
+        return false;
+      case ANY:
+        return true;
+      case OBJECT_OR_ARRAY_ONLY:
+        return isJsonObjectOrArray(data);
+      default:
+        throw new IllegalStateException("Unexpected value: " + convertJson);
+    }
+  }
```

Comment:
`IllegalStateException` may only be thrown here if there is a bug in the MongoDB Spark Connector: we introduced a new `enum` constant of type `WriteConfig.ConvertJson`, but forgot to add a `case` for it here. In the MongoDB Java Driver we throw `AssertionError` such cases (see `com.mongodb.assertions.Assertions`) to make it clear that there is a driver bug. Do we want to start using the same approach in the MongoDB Spark Connector?

Reply:
Done